### PR TITLE
Orchestrated build changes

### PR DESCRIPTION
**Pull request by jaredpar**
These are the source changes which go with the orchestrated build definition changes. The build definition needs to change before merging this PR.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @mmitche

**Comment by jcouv** [comment]
Do we have any overview of the orchestrated build design, for context?

**Comment by tmat** [comment]
👍

File: build/scripts/build.ps1 [hunk 1]
```diff
@@ -513,9 +514,12 @@ function Run-SignTool() {
     try {
         $signTool = Join-Path (Get-PackageDir "RoslynTools.Microsoft.SignTool") "tools\SignTool.exe"
         $signToolArgs = "-msbuildPath `"$msbuild`""
-        if (-not $official) {
-            $signToolArgs += " -test"
+        switch ($signType) {
+            "real" { break; }
+            "test" { $signToolArgs += " -testSign"; break; }
+            default { $signToolArgs += " -test"; break; }
```
Comment (jasonmalinowski) [inline build/scripts/build.ps1]: I must admit this switch doesn't make sense to me. I get the test to testSign line, but default just throws in -test? Is that a typo?

Comment (jasonmalinowski) [reply build/scripts/build.ps1]: (I suspect my real statement here is "please add comments")

Comment (jaredpar) [reply build/scripts/build.ps1]: There are two different features here:

- "-test": this is similar to "-n" in many unix commands. It runs the SignTool in verification mode. No signing is done here. Instead it simply verifies the consistentcy of the build output with the information in the configuration file. 
- "-testSign": the signing mechanism is test signing, not real signing.

In retrospect I should have called "-test" "-verifyOnly". May still change it but worry that will cause a lot of churn.

**Pull request merged**